modules/build/src/main/scala/scala/build/CrossSources.scala

```diff
@@ -152,6 +152,31 @@ object CrossSources {
         .map(_.flatten)
 
     val preprocessedInputFromArgs = value(preprocessSources(inputs.flattened()))
+    val preprocessedWithUsingDirs = preprocessedInputFromArgs.filter(_.directivesLocation.isDefined)
+
+    if (preprocessedWithUsingDirs.length > 1) {
+      val projectFilePath = inputs.elements.projectSettingsFiles.headOption match
+        case Some(s) => s.path
+        case _       => inputs.workspace / "project.scala"
+      preprocessedWithUsingDirs
+        .filter(_.scopePath != ScopePath.fromPath(projectFilePath))
+        .foreach { source =>
+          source.directivesLocation match
+            case Some(location) =>
+              val dirsScope = Seq(
+                location.codeDirectives.line,
+                location.specialCommentDirectives.line,
+                location.plainCommentDirectives.line
+              ).max
+              val path = source.scopePath.root match
+                case Right(base: os.Path) => os.Path(source.scopePath.path, base).toString
+                case Left(str: String)    => str + source.scopePath.path.toString
+              logger.diagnostic(
```

Comment:
Try using the `positions` arguments in `logger.diagnostic` and check how it is displayed in Metals. If you set the `positions` arguments correctly, the metals should show a warning message in the file.

It would be helpful in the future, if we decided here to add actionable diagnostic.

Reply:
I just committed changes, which use positions in logger.diagnostics. It works well with metals ide.